scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -2281,14 +2281,19 @@ def _get_containers(
                         )
                     except k8s_utils.K8sApiException as e:
                         # If the pod details cannot be retrieved from the K8s API:
-                        # 404 - log warning
+                        # 404 - log warning, include based on k8s_include_by_default
                         # Otherwise (401 as per K8s Api documentation) - log error
                         # Exclude the pod
                         if e.status_code == 404:
-                            global_log.warning(
-                                "Pod %s/%s not found in K8s API. Excluding pod from collection."
-                                % (pod_namespace, pod_name),
-                                exc_info=e,
+                            if k8s_include_by_default:
+                                include_message = "Including pod based on SCALYR_K8S_INCLUDE_ALL_CONTAINERS=true."
+                            else:
+                                include_message = "Excluding pod based on SCALYR_K8S_INCLUDE_ALL_CONTAINERS=false."
+                            global_log.info(
+                                "Pod %s/%s not found in K8s API. %s"
+                                % (pod_namespace, pod_name, include_message),
+                                limit_once_per_x_secs=300,
+                                limit_key=pod_name,
```

Comment:
Limit keys needs to to be globally unique so to avoid collisions / conflicts, it would be better to call it something along the lines of `k8s-api-pod-not-found-{pod_name}`.

Having said that, we should also confirm with the logger limit implementation to make sure those entries eventually get purged otherwise we could have problems with memory usage since this is a high cardinality value. If they don't, we may need to use a different key.

Reply:
Removing the logging key after discussion with @tomaz-s1 , because of the logging framework not implementing eviction of old timestamp records.